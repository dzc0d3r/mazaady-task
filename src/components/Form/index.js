import Form from "./Form"
import Loading from "./Loading"

export { Form, Loading }
