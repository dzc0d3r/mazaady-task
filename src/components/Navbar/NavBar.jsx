"use client"
import Image from "next/image"
import Link from "next/link"
import { useState } from "react"
import { HiMenu } from "react-icons/hi"
import { MdOutlineClose } from "react-icons/md"
import { PiGlobe } from "react-icons/pi"
import { IoChatbubbleOutline } from "react-icons/io5"
import { GoBell } from "react-icons/go"
import { FiPlusCircle } from "react-icons/fi"

const NavBar = () => {
  const [navbar, setNavbar] = useState(false)
  return (
    <div className="sticky top-0 z-50 w-full bg-white">
      <div className="container mx-auto max-w-7xl bg-white">
        <div className=" relative flex h-16 flex-row justify-between bg-white md:items-center md:px-8">
          <div className=" md:flex md:flex-row md:items-center">
            <div className=" flex items-center py-3 md:py-0 ">
              <div className="flex items-center gap-x-4 md:hidden ">
                <button
                  className="mx-2 rounded-md p-2 text-gray-500  outline-none"
                  onClick={() => setNavbar(!navbar)}
                >
                  {navbar ? (
                    <>
                      <MdOutlineClose className="m-0 p-0 text-3xl text-gray-300" />
                    </>
                  ) : (
                    <HiMenu className="m-0 p-0 text-3xl text-gray-300" />
                  )}
                </button>
              </div>
              {/* LOGO */}
              <Link href="/">
                <Image
                  src="logo.svg"
                  width={100}
                  height={100}
                  alt="logo"
                  className="mr-10 h-max w-24 md:h-28 md:w-28"
                />
              </Link>
            </div>

            <div
              className={`md:flex ${navbar ? "block w-screen md:w-auto lg:w-0" : "hidden"}`}
            >
              <ul className="-mt-2 flex h-screen flex-col items-center justify-start space-y-2 bg-white pt-10 capitalize md:mt-0 md:flex md:h-auto md:flex-row md:items-center md:justify-center  md:space-x-10 md:space-y-0 md:pt-0">
                <li
                  className={`active mx-auto block w-48 rounded-full py-2 text-center text-lg text-gray-500 md:w-auto md:py-0
                                     `}
                >
                  <Link href="/" onClick={() => setNavbar(!navbar)}>
                    home
                  </Link>
                </li>
                <li className="mx-auto block w-48 rounded-full py-2 text-center text-lg text-gray-500 md:w-auto md:py-0">
                  <Link href="/" onClick={() => setNavbar(!navbar)}>
                    blogs
                  </Link>
                </li>

                <li className="mx-auto block w-48 rounded-full py-2 text-center text-lg text-gray-500 md:w-auto md:py-0">
                  <Link href="/contact" onClick={() => setNavbar(!navbar)}>
                    gifts
                  </Link>
                </li>
              </ul>
            </div>
          </div>

          <div>
            <div className="group flex h-full w-full ">
              <div className="mx-5 flex items-center group-hover:cursor-pointer md:divide-x-2 md:divide-gray-100 ">
                {/* Conversations */}

                <div className="p-2">
                  <IoChatbubbleOutline className="m-1 rotate-[280deg] text-xl text-gray-700" />
                </div>

                {/* Notifications */}
                <div className="p-2">
                  <GoBell className="m-1 text-xl text-gray-700" />
                </div>
                {/* Profile image */}

                <div className="mx-auto w-full px-3">
                  <Image
                    src="profile.svg"
                    height={50}
                    width={50}
                    alt="Profile image"
                    className="mx-auto h-11 w-11 rounded-full"
                  ></Image>
                </div>

                {/* Add Porduct button */}
              </div>

              <Link href="/add-product">
                <button className="hidden w-full flex-row items-center justify-center gap-x-2 rounded-xl bg-gradient-to-r from-primary to-orange-600 px-1 py-1 text-sm font-semibold text-white hover:to-orange-500 disabled:pointer-events-none disabled:opacity-50  md:flex lg:w-fit lg:px-4 lg:py-2 lg:text-base">
                  <FiPlusCircle className="text-2xl" />
                  Add New Product
                </button>
              </Link>
              <div className="mx-5 hidden grid-cols-2 items-center md:grid lg:divide-x-2 ">
                <button className="mx-auto flex w-20  items-center  justify-center px-1 text-xl md:w-auto">
                  <PiGlobe className="text-2xl" />
                </button>

                <div className="hidden w-full px-1 text-end lg:flex">EN</div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

export default NavBar
