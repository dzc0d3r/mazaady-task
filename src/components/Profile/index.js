import ProfileCard from "./ProfileCard"
import QRCard from "./QRCard"
import Products from "./Products"
import SingleProduct from "./SingleProduct"

export { ProfileCard, QRCard, Products, SingleProduct }
