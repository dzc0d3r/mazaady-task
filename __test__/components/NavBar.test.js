import renderer from "react-test-renderer"

import { NavBar } from "@/components/Navbar"

describe("Navar tests", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<NavBar />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
