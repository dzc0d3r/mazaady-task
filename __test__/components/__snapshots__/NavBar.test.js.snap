// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navar tests renders correctly 1`] = `
<div
  className="w-full bg-white sticky top-0 z-50"
>
  <div
    className="container mx-auto max-w-7xl bg-white"
  >
    <div
      className=" relative h-16 justify-between bg-white flex flex-row md:items-center md:px-8"
    >
      <div
        className=" md:flex md:flex-row md:items-center"
      >
        <div
          className=" flex items-center py-3 md:py-0 "
        >
          <div
            className="flex items-center gap-x-4 md:hidden "
          >
            <button
              className="mx-2 rounded-md p-2 text-gray-500  outline-none"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="m-0 p-0 text-3xl text-gray-300"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  {
                    "color": undefined,
                  }
                }
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clipRule="evenodd"
                  d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
                  fillRule="evenodd"
                />
              </svg>
            </button>
          </div>
          <a
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            <img
              alt="logo"
              className="mr-10 h-max w-24 md:h-28 md:w-28"
              data-nimg="1"
              decoding="async"
              height={100}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="logo.svg"
              style={
                {
                  "color": "transparent",
                }
              }
              width={100}
            />
          </a>
        </div>
        <div
          className="md:block  hidden"
        >
          <ul
            className="-mt-2 flex h-screen flex-col items-center justify-start space-y-2 bg-white pt-10 capitalize md:mt-0 md:flex md:h-auto md:flex-row md:items-center md:justify-center  md:space-x-10 md:space-y-0 md:pt-0"
          >
            <li
              className="active mx-auto block w-48 rounded-full py-2 text-center text-lg text-gray-500 md:w-auto md:py-0
                                     "
            >
              <a
                href="/"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                home
              </a>
            </li>
            <li
              className="mx-auto block w-48 rounded-full py-2 text-center text-lg text-gray-500 md:w-auto md:py-0"
            >
              <a
                href="/"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                blogs
              </a>
            </li>
            <li
              className="mx-auto block w-48 rounded-full py-2 text-center text-lg text-gray-500 md:w-auto md:py-0"
            >
              <a
                href="/contact"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                gifts
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div>
        <div
          className="h-full w-full flex group "
        >
          <div
            className="group-hover:cursor-pointer mx-5 flex items-center md:divide-x-2 md:divide-gray-100 "
          >
            <div
              className="p-2"
            >
              <svg
                className="m-1 rotate-[280deg] text-xl text-gray-700"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  {
                    "color": undefined,
                  }
                }
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M87.49 380c1.19-4.38-1.44-10.47-3.95-14.86a44.86 44.86 0 0 0-2.54-3.8 199.81 199.81 0 0 1-33-110C47.65 139.09 140.73 48 255.83 48 356.21 48 440 117.54 459.58 209.85a199 199 0 0 1 4.42 41.64c0 112.41-89.49 204.93-204.59 204.93-18.3 0-43-4.6-56.47-8.37s-26.92-8.77-30.39-10.11a31.09 31.09 0 0 0-11.12-2.07 30.71 30.71 0 0 0-12.09 2.43l-67.83 24.48a16 16 0 0 1-4.67 1.22 9.6 9.6 0 0 1-9.57-9.74 15.85 15.85 0 0 1 .6-3.29z"
                  fill="none"
                  strokeLinecap="round"
                  strokeMiterlimit="10"
                  strokeWidth="32"
                />
              </svg>
            </div>
            <div
              className="p-2"
            >
              <svg
                className="m-1 text-xl text-gray-700"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 1c3.681 0 7 2.565 7 6v4.539c0 .642.189 1.269.545 1.803l2.2 3.298A1.517 1.517 0 0 1 20.482 19H15.5a3.5 3.5 0 1 1-7 0H3.519a1.518 1.518 0 0 1-1.265-2.359l2.2-3.299A3.25 3.25 0 0 0 5 11.539V7c0-3.435 3.318-6 7-6ZM6.5 7v4.539a4.75 4.75 0 0 1-.797 2.635l-2.2 3.298-.003.01.001.007.004.006.006.004.007.001h16.964l.007-.001.006-.004.004-.006.001-.006a.017.017 0 0 0-.003-.01l-2.199-3.299a4.753 4.753 0 0 1-.798-2.635V7c0-2.364-2.383-4.5-5.5-4.5S6.5 4.636 6.5 7ZM14 19h-4a2 2 0 1 0 4 0Z"
                />
              </svg>
            </div>
            <div
              className="mx-auto w-full px-3"
            >
              <img
                alt="Profile image"
                className="mx-auto h-11 w-11 rounded-full"
                data-nimg="1"
                decoding="async"
                height={50}
                loading="lazy"
                onError={[Function]}
                onLoad={[Function]}
                src="profile.svg"
                style={
                  {
                    "color": "transparent",
                  }
                }
                width={50}
              />
            </div>
          </div>
          <button
            className="hidden w-full lg:w-fit md:flex flex-row justify-center items-center gap-x-2 rounded-xl bg-gradient-to-r from-primary to-orange-600 lg:px-4 lg:py-2 px-1 py-1 lg:text-base text-sm  font-semibold text-white hover:to-orange-500 disabled:pointer-events-none disabled:opacity-50"
          >
            <svg
              className="text-2xl"
              fill="none"
              height="1em"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <line
                x1="12"
                x2="12"
                y1="8"
                y2="16"
              />
              <line
                x1="8"
                x2="16"
                y1="12"
                y2="12"
              />
            </svg>
            Add New Product
          </button>
          <div
            className="mx-5 hidden md:grid grid-cols-2 items-center lg:divide-x-2 "
          >
            <button
              className="mx-auto flex w-20  items-center  justify-center text-xl md:w-auto px-1"
            >
              <svg
                className="text-2xl"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  {
                    "color": undefined,
                  }
                }
                viewBox="0 0 256 256"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24ZM101.63,168h52.74C149,186.34,140,202.87,128,215.89,116,202.87,107,186.34,101.63,168ZM98,152a145.72,145.72,0,0,1,0-48h60a145.72,145.72,0,0,1,0,48ZM40,128a87.61,87.61,0,0,1,3.33-24H81.79a161.79,161.79,0,0,0,0,48H43.33A87.61,87.61,0,0,1,40,128ZM154.37,88H101.63C107,69.66,116,53.13,128,40.11,140,53.13,149,69.66,154.37,88Zm19.84,16h38.46a88.15,88.15,0,0,1,0,48H174.21a161.79,161.79,0,0,0,0-48Zm32.16-16H170.94a142.39,142.39,0,0,0-20.26-45A88.37,88.37,0,0,1,206.37,88ZM105.32,43A142.39,142.39,0,0,0,85.06,88H49.63A88.37,88.37,0,0,1,105.32,43ZM49.63,168H85.06a142.39,142.39,0,0,0,20.26,45A88.37,88.37,0,0,1,49.63,168Zm101.05,45a142.39,142.39,0,0,0,20.26-45h35.43A88.37,88.37,0,0,1,150.68,213Z"
                />
              </svg>
            </button>
            <div
              className="px-1 w-full text-end lg:flex hidden"
            >
              EN
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
